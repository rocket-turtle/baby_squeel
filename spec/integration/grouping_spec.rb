require 'spec_helper'

describe '#grouping' do
  it 'groups on a column' do
    relation = Post.selecting { id.count }.grouping { author_id }

    expect(relation).to match_sql_snapshot
  end

  it 'groups on a calculation' do
    relation = Post.selecting { id.count }.grouping {
      (author_id + 1) + 5
    }

    expect(relation).to match_sql_snapshot
  end

  it 'groups on an association' do
    relation = Post.joins(:author).selecting { id.count }.grouping { author.id }

    expect(relation).to match_sql_snapshot
  end

  it 'groups on an aliased association' do
    relation = Post.joining { author.posts }
                   .selecting { author.posts.id.count }
                   .grouping { author.posts.author_id }

    expect(relation).to match_sql_snapshot(variants: ['8.1'])
  end
end
