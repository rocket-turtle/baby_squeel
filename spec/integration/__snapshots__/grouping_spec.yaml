---
"#grouping groups on a column 1": SELECT COUNT("posts"."id") FROM "posts" GROUP BY
  "posts"."author_id"
"#grouping groups on a calculation 1": SELECT COUNT("posts"."id") FROM "posts" GROUP
  BY (("posts"."author_id" + 1) + 5)
"#grouping groups on an association 1": SELECT COUNT("posts"."id") FROM "posts" INNER
  JOIN "authors" ON "authors"."id" = "posts"."author_id" GROUP BY "authors"."id"
"#grouping groups on an aliased association 1": SELECT COUNT("posts_authors"."id")
  FROM "posts" INNER JOIN "authors" ON "authors"."id" = "posts"."author_id" INNER
  JOIN "posts" "posts_authors" ON "posts_authors"."author_id" = "authors"."id" GROUP
  BY "posts_authors"."author_id"
"#grouping groups on an aliased association 1 (Active Record: v8.1)": SELECT COUNT("posts_authors"."id")
  FROM "posts" INNER JOIN "authors" ON "authors"."id" = "posts"."author_id" INNER
  JOIN "posts" AS "posts_authors" ON "posts_authors"."author_id" = "authors"."id"
  GROUP BY "posts_authors"."author_id"
